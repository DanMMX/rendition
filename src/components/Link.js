const get = require('lodash/get')
const omit = require('lodash/omit')
import * as React from 'react'
import styled from 'styled-components'
import { color, fontSize, space } from 'styled-system'
import { align, bold, caps } from './Txt'
import asRendition from '../asRendition'
import { darken, monospace } from '../utils'

let Base = styled.a`
  ${align}
  ${color}
  ${fontSize}
  ${monospace};
  ${space}

  ${caps}
  ${bold}

  text-decoration: ${props => props.decor || 'none'};
  cursor: ${props => (props.disabled ? 'not-allowed' : 'pointer')};
  opacity: ${props => (props.disabled ? 0.65 : 1)};
  display: inline-block;

  &:active,
  &:hover {
    color: ${props =>
    !props.disabled &&
      darken(get(props.theme.colors, props.color) || props.color)};
  }
`

const Link = ({ is, blank, ...props }) => {
  if (is) {
    Base = Base.withComponent(is)
  }
  if (props.disabled) {
    props = omit(props, 'href')
  }
  return (
    <Base {...props} rel={blank && 'noopener'} target={blank && '_blank'} />
  )
}

Link.defaultProps = {
  is: 'a',
  color: 'primary.main'
}

export default asRendition(Link)
