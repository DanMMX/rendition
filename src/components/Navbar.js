import * as React from 'react'
const MenuIcon = require('react-icons/lib/md/dehaze')
const CloseIcon = require('react-icons/lib/md/close')
import styled, { withTheme } from 'styled-components'
import { responsiveStyle } from 'styled-system'
import { Flex, Box } from './Grid'
import Container from './Container'

const display = responsiveStyle('display', 'display')
const maxHeight = responsiveStyle('max-height', 'maxHeight')

const BrandBox = styled(Box)`
  display: flex;
  height: auto;
  min-width: 150px;
  align-self: center;
`

const IconBox = styled(Box)`
  ${display} align-self: center;
  cursor: pointer;
`

const MenuBox = styled(Flex)`
  ${display} ${maxHeight};
  text-align: center;
  transition: max-height 0.4s ease-in-out;
  align-self: center;
  overflow: hidden;

  a {
    width: 100%;
    display: inline-block;
  }
`

class Navbar extends React.Component {
  constructor (props) {
    super(props)
    this.state = {
      open: false
    }
  }

  toggle () {
    this.setState({
      open: !this.state.open
    })
  }

  render () {
    const { children, brand, ...props } = this.props
    return (
      <Box {...props}>
        <Container>
          <Flex direction={['column', 'row']}>
            <Flex>
              <BrandBox p={2}>{brand}</BrandBox>
              <IconBox
                onClick={() => this.toggle()}
                ml={['auto', 0]}
                p={2}
                display={['block', 'none']}
              >
                {this.state.open ? <CloseIcon /> : <MenuIcon />}
              </IconBox>
            </Flex>
            <MenuBox
              maxHeight={[this.state.open ? '100vh' : '0vh', 'none']}
              width={['100%', 'auto']}
              ml={[0, 'auto']}
            >
              <Flex width={['100%', 'auto']} direction={['column', 'row']}>
                {children.map((child, i) => {
                  return (
                    <Box
                      onClick={() => this.toggle()}
                      width={['100%', 'auto']}
                      p={2}
                      key={i}
                    >
                      {child}
                    </Box>
                  )
                })}
              </Flex>
            </MenuBox>
          </Flex>
        </Container>
      </Box>
    )
  }
}

Navbar.defaultProps = {
  color: 'white',
  bg: 'gray.dark'
}

export default withTheme(Navbar)
