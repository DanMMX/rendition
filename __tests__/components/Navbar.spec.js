/* globals expect, test */
import React from 'react'
import renderer from 'react-test-renderer'
import Link from '../../src/components/Link'
import Provider from '../../src/components/Provider'
import Navbar from '../../src/components/Navbar'

test('Navbar renders correctly', () => {
  const component = renderer.create(
    <Provider>
      <Navbar color='white'>
        <Link color='white' href={'/docs/'}>
          Docs
        </Link>
        <Link color='white' href={'/changelog/'}>
          changelog
        </Link>
        <Link color='white' href={'/gitter/'}>
          gitter
        </Link>
      </Navbar>
    </Provider>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
