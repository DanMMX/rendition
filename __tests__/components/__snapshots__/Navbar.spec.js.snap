// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders correctly 1`] = `
<div
  className="sc-htoDjs bSlTVa"
>
  <div
    className="sc-jTzLTM SofDJ sc-gqjmRU gYzTfQ sc-dnqmqq fozHQE"
  >
    <div
      className="sc-fjdhpX gKwwVP"
    >
      <div
        className="sc-VigVT hegYug sc-gZMcBi eQxJRS sc-iwsKbI gLKxKc"
        direction={
          Array [
            "column",
            "row",
          ]
        }
      >
        <div
          className="sc-VigVT hegYug sc-gZMcBi eQxJRS sc-iwsKbI gLKxKc"
        >
          <div
            className="sc-jzJRlG dVdezz sc-jTzLTM dqvzgf sc-gqjmRU jowXRS sc-dnqmqq gGbYAg"
          />
          <div
            className="sc-cSHVUG iCVeHJ sc-jTzLTM bxmMAi sc-gqjmRU jowXRS sc-dnqmqq fiBbgU"
            display={
              Array [
                "block",
                "none",
              ]
            }
            onClick={[Function]}
          >
            <svg
              fill="currentColor"
              height="1em"
              preserveAspectRatio="xMidYMid meet"
              style={
                Object {
                  "color": undefined,
                  "verticalAlign": "middle",
                }
              }
              viewBox="0 0 40 40"
              width="1em"
            >
              <g>
                <path
                  d="m3.4 9.1h33.2v3.4h-33.2v-3.4z m0 8.4h33.2v3.4h-33.2v-3.4z m0 8.4h33.2v3.2h-33.2v-3.2z"
                />
              </g>
            </svg>
          </div>
        </div>
        <div
          className="sc-kAzzGY buEXUc sc-VigVT giMOMu sc-gZMcBi eQxJRS sc-iwsKbI fGhqpY"
          maxHeight={
            Array [
              "0vh",
              "none",
            ]
          }
        >
          <div
            className="sc-VigVT iLaKEt sc-gZMcBi eQxJRS sc-iwsKbI hCcawn"
            direction={
              Array [
                "column",
                "row",
              ]
            }
          >
            <div
              className="sc-jTzLTM hpYooM sc-gqjmRU jowXRS sc-dnqmqq llRrFg"
              onClick={[Function]}
            >
              <a
                className="sc-gzVnrw cpqKMG sc-chPdSV goMgoP"
                color="white"
                href="/docs/"
                rel={undefined}
                target={undefined}
              >
                Docs
              </a>
            </div>
            <div
              className="sc-jTzLTM hpYooM sc-gqjmRU jowXRS sc-dnqmqq llRrFg"
              onClick={[Function]}
            >
              <a
                className="sc-gzVnrw cpqKMG sc-kgoBCf gXqCXL"
                color="white"
                href="/changelog/"
                rel={undefined}
                target={undefined}
              >
                changelog
              </a>
            </div>
            <div
              className="sc-jTzLTM hpYooM sc-gqjmRU jowXRS sc-dnqmqq llRrFg"
              onClick={[Function]}
            >
              <a
                className="sc-gzVnrw cpqKMG sc-kGXeez jASSHv"
                color="white"
                href="/gitter/"
                rel={undefined}
                target={undefined}
              >
                gitter
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
